scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+                    continue
+                else:
+                    self._send_reply(self._make_fail(self._error))
+                    break
+
+            if command == "stop":
+                self._logger.info("Stop command received")
+                self._stop_nodes()
+                self._send_reply()
+                break
+
+            self._logger.info(f"Executing command {command}")
+
+            try:
+                args = _parse_args(self._app_name, command.split(' '))
+                op = args.operation.replace('-', '_')
+                if not hasattr(self, op):
+                    raise RuntimeError(f"Invalid supervisor command: {op}")
+                args_dict = vars(args)
+                del args_dict["operation"]
+                self._send_reply(getattr(self, op)(**args_dict))
+                self._logger.info("Command execution successful")
+            except Exception as e:
+                self._logger.error(f"Command execution failed: {traceback.format_exc()}")
+                self._send_reply(self._make_fail(e))
+
+        self._logger.info("Listen loop terminated")
+        time.sleep(0.2) # give time to the client to read the last response
+
+
+def _get_fifo_paths(cluster_directory):
+    sin = os.path.join(cluster_directory, FIFO_IN)
+    sout = os.path.join(cluster_directory, FIFO_OUT)
+    return sin, sout
+
+
+class SupervisorClient:
+    """Local client, for the supervisor
+
+    Allows initializing a connection and sending commands to the supervisor.
+    """
+
+    def __init__(self, cluster_directory):
+        """Initialize the client for a supervisor running in the specified cluster directory"""
+        self.supervisor_in_fifo, self.supervisor_out_fifo = _get_fifo_paths(cluster_directory)
+
+    def send_command(self, argv):
+        """Send commands to the supervisor
+
+        Only node commands are supported, e.g. commands ending with the *-node
+        prefix as opposed to those with a *-cluster prefix.
+        Pass the full command-line in the argv parameter, e.g.:
+
+            client = sscm.SupervisorClient("/path/to/cluster_dir")
+            client.send_command(["start-node", "-i1"])
+
+        """
+        if not os.path.exists(self.supervisor_in_fifo) or not os.path.exists(self.supervisor_out_fifo):
+            raise RuntimeError(f"Failed to find supervisor input/output FIFO at {self.supervisor_in_fifo} and {self.supervisor_out_fifo}, is the supervisor running?")
+
+        with open(self.supervisor_in_fifo, "w") as pout:
+            with open(self.supervisor_out_fifo, "r") as pin:
+                pout.write(' '.join(argv) + '\n')
+                pout.flush()
+
+                res = json.loads(pin.readline())
+                if res["success"]:
+                    if res["content"]:
+                        return res["content"]
+                    return None
+
+                raise RuntimeError(f"Operation failed, supervisor responed with: {res['content']}")
```

Comment:
since we are always interested in `res['content']`, could store it locally, less repeatings this way:

```py
response = res["concent"]
if res["success"]:
    return response or None
```
but why do we want to normalize it to `None`? i think the caller always want to print it. so returning an empty string should be fine, no? and would be simpler if we wanted to annotate the return type. (i'd tempted to do so =)

Reply:
Done in v2.